ndcube/asdf/converters/ndcube_converter.py

```diff
@@ -0,0 +1,50 @@
+import warnings
+
+from asdf.extension import Converter
+
+
+class NDCubeConverter(Converter):
+    tags = ["tag:sunpy.org:ndcube/ndcube/ndcube-*"]
+    types = ["ndcube.ndcube.NDCube"]
+
+    def from_yaml_tree(self, node, tag, ctx):
+        from ndcube.ndcube import NDCube
+
+        ndcube = NDCube(node["data"], node["wcs"])
+        ndcube._extra_coords = node["extra_coords"]
+        ndcube._global_coords = node["global_coords"]
```

Comment:
These two are not required in the schema, so I think they should be optional:
```suggestion
        if "extra_coords" in node:
            ndcube._extra_coords = node["extra_coords"]
        if "global_coords" in node:
            ndcube._global_coords = node["global_coords"]
```